ext/standard/streamsfuncs.c

```diff
@@ -616,93 +616,139 @@ PHP_FUNCTION(stream_get_wrappers)
 }
 /* }}} */
 
-/* {{{ stream_select related functions */
-static int stream_array_to_fd_set(zval *stream_array, fd_set *fds, php_socket_t *max_fd)
-{
-	zval *elem;
-	php_stream *stream;
-	int cnt = 0;
+/* {{{ fd_bigset related macros and typedefs to remove FD_SETSIZE limitation in stream_select */
+typedef struct {
+    char *fds_bits;
+    size_t size;
+} fd_bigset;
+#define FD_BIGSET_ENSURE_CAPACITY(fd, set) \
+    if (UNEXPECTED(((fd) / 8 >= (set)->size))) { \
+        fd_bigset_double_size(set); \
+    }
+#define FD_BIGSET_ZERO(set, num_fds) do { \
+    (set)->size = (num_fds + 7) / 8; \
+    (set)->fds_bits = (char *)ecalloc((set)->size, sizeof(char)); \
+} while (0)
+#define FD_BIGSET_SET(fd, set) do { \
+	FD_BIGSET_ENSURE_CAPACITY(fd, set) \
+	(set)->fds_bits[(fd) / 8] |= (1 << ((fd) % 8)); \
+} while (0)
+#define FD_BIGSET_ISSET(fd, set) (((fd) / 8 < (set)->size) ? ((set)->fds_bits[(fd) / 8] & (1 << ((fd) % 8))) : 0)
+#define FD_BIGSET_CLR(fd, set) do { \
+    if ((fd) / 8 < (set)->size) { \
+        (set)->fds_bits[(fd) / 8] &= ~(1 << ((fd) % 8)); \
+    } \
+} while (0)
+#define FD_BIGSET_FREE(set) do { \
+    if ((set)->fds_bits) { \
+        efree((set)->fds_bits); \
+        (set)->fds_bits = NULL; \
+    } \
+    (set)->size = 0; \
+} while (0)
+
+/* {{{ Function to double the size of an fd_bigset */
+static void fd_bigset_double_size(fd_bigset *set) {
+	ZEND_ASSERT(set && set->fds_bits);
+
+    size_t old_size = set->size;
+    size_t new_size = old_size * 2;
```

Comment:
Can you please fix coding style - we use tabs, not spaces. This is making the diff bigger and harder to review.